econml/_ortho_learner.py

```diff
@@ -540,15 +558,70 @@ def fit(self, Y, T, X=None, W=None, Z=None, *, sample_weight=None, sample_var=No
         Y, T, X, W, Z, sample_weight, sample_var, groups = check_input_arrays(
             Y, T, X, W, Z, sample_weight, sample_var, groups)
         self._check_input_dims(Y, T, X, W, Z, sample_weight, sample_var, groups)
-        nuisances, fitted_inds = self._fit_nuisances(Y, T, X, W, Z, sample_weight=sample_weight, groups=groups)
-        self._fit_final(self._subinds_check_none(Y, fitted_inds),
-                        self._subinds_check_none(T, fitted_inds),
-                        X=self._subinds_check_none(X, fitted_inds),
-                        W=self._subinds_check_none(W, fitted_inds),
-                        Z=self._subinds_check_none(Z, fitted_inds),
-                        nuisances=tuple([self._subinds_check_none(nuis, fitted_inds) for nuis in nuisances]),
-                        sample_weight=self._subinds_check_none(sample_weight, fitted_inds),
-                        sample_var=self._subinds_check_none(sample_var, fitted_inds))
+
+        all_nuisances = []
+        fitted_inds = None
+
+        for _ in range(monte_carlo_iterations or 1):
+            nuisances, new_inds = self._fit_nuisances(Y, T, X, W, Z, sample_weight=sample_weight, groups=groups)
+            all_nuisances.append(nuisances)
+            if fitted_inds is None:
+                fitted_inds = new_inds
+            elif not np.array_equal(fitted_inds, new_inds):
+                raise AttributeError("Different indices were fit by different folds, so they cannot be aggregated")
+
+        if monte_carlo_iterations is not None:
+            # TODO: support different ways to aggregate, like median?
+            nuisances = np.mean(np.array(all_nuisances), axis=0)
+
+        Y, T, X, W, Z, sample_weight, sample_var = (self._subinds_check_none(arr, fitted_inds)
+                                                    for arr in (Y, T, X, W, Z, sample_weight, sample_var))
+        nuisances = tuple([self._subinds_check_none(nuis, fitted_inds) for nuis in nuisances])
+        self._cached_values = CachedValues(nuisances=nuisances,
+                                           Y=Y, T=T, X=X, W=W, Z=Z,
+                                           sample_weight=sample_weight,
+                                           sample_var=sample_var) if cache_values else None
+        self._fit_final(Y=Y, T=T, X=X, W=W, Z=Z,
+                        nuisances=nuisances,
+                        sample_weight=sample_weight,
+                        sample_var=sample_var)
+
+        self._cache_invalid_message = None
+        return self
+
+    # TODO: this doesn't currently allow inference, because that is handled by the wrap_fit attribute
```

Comment:
So is inference the same object as the one called by fit?

I guess this is also related to the fact that the inference object does not have the new model final. I suspect this might need to be solved simultaneously (i.e. propagate model final to the inference object and allow refit to take as input "inference=...")